### ISSUE TITLE
New changes in kubebuilder causing issue in SDK testcases

### ISSUE THREAD

**Issue by laxmikantbpandhare** [operator-framework/operator-sdk#6046]
#### Describe the problem you need a feature to resolve.

kubebuilder scaffolding logic got changed

#### Describe the solution you'd like.

go lang

Before kb changes 

``` $ operator-sdk init --domain example.com --repo github.com/example/memcached-operator
Writing kustomize manifests for you to edit...
Writing scaffold for you to edit...
Get controller runtime:
$ go get sigs.k8s.io/controller-runtime@v0.13.0
Update dependencies:
$ go mod tidy
Next: define a resource with:
$ operator-sdk create api
laxmikantbhaskarpandhare@lpandhar-mac confog-manager % tree
.
├── Dockerfile
├── Makefile
├── PROJECT
├── README.md
├── config
│   ├── default
│   │   ├── kustomization.yaml
│   │   ├── manager_auth_proxy_patch.yaml
│   │   └── manager_config_patch.yaml
│   ├── manager
│   │   ├── controller_manager_config.yaml
│   │   ├── kustomization.yaml
│   │   └── manager.yaml
│   ├── manifests
│   │   └── kustomization.yaml
│   ├── prometheus
│   │   ├── kustomization.yaml
│   │   └── monitor.yaml
│   ├── rbac
│   │   ├── auth_proxy_client_clusterrole.yaml
│   │   ├── auth_proxy_role.yaml
│   │   ├── auth_proxy_role_binding.yaml
│   │   ├── auth_proxy_service.yaml
│   │   ├── kustomization.yaml
│   │   ├── leader_election_role.yaml
│   │   ├── leader_election_role_binding.yaml
│   │   ├── role_binding.yaml
│   │   └── service_account.yaml
│   └── scorecard
│       ├── bases
│       │   └── config.yaml
│       ├── kustomization.yaml
│       └── patches
│           ├── basic.config.yaml
│           └── olm.config.yaml
├── go.mod
├── go.sum
├── hack
│   └── boilerplate.go.txt
└── main.go

10 directories, 30 files
```

After

```
.
├── Dockerfile
├── Makefile
├── PROJECT
├── README.md
├── config
│   ├── default
│   │   ├── kustomization.yaml
│   │   ├── manager_auth_proxy_patch.yaml
│   │   └── manager_config_patch.yaml
│   ├── manager
│   │   ├── kustomization.yaml
│   │   └── manager.yaml
│   ├── manifests
│   │   └── kustomization.yaml
│   ├── prometheus
│   │   ├── kustomization.yaml
│   │   └── monitor.yaml
│   ├── rbac
│   │   ├── auth_proxy_client_clusterrole.yaml
│   │   ├── auth_proxy_role.yaml
│   │   ├── auth_proxy_role_binding.yaml
│   │   ├── auth_proxy_service.yaml
│   │   ├── kustomization.yaml
│   │   ├── leader_election_role.yaml
│   │   ├── leader_election_role_binding.yaml
│   │   ├── role_binding.yaml
│   │   └── service_account.yaml
│   └── scorecard
│       ├── bases
│       │   └── config.yaml
│       ├── kustomization.yaml
│       └── patches
│           ├── basic.config.yaml
│           └── olm.config.yaml
├── go.mod
├── go.sum
├── hack
│   └── boilerplate.go.txt
└── main.go
```

``config/manager/controller_manager_config.yaml` will be scaffolded only after `--component-config` flag is enabled.

The Kubebuilder issue is here - https://github.com/kubernetes-sigs/kubebuilder/issues/2782
PR for the same: https://github.com/kubernetes-sigs/kubebuilder/pull/2826

**Comment by laxmikantbpandhare** [comment]
As part of this issue, we need to modify test cases and docs from `operator-sdk`.

**Comment by laxmikantbpandhare** [comment]
```
ERRO[0001] error creating the project: operator-sdk init --plugins helm --domain example.com --group cache --version v1alpha1 --kind Memcached --helm-chart ../../../hack/generate/samples/internal/helm/testdata/memcached-0.0.2.tgz failed with error: (exit status 1) Writing kustomize manifests for you to edit...
Error: failed to initialize project: unable to scaffold with "base.helm.sdk.operatorframework.io/v1": error updating init manifests: stat config/manager/controller_manager_config.yaml: no such file or directory
```

**Comment by laxmikantbpandhare** [comment]
KB issue - https://github.com/kubernetes-sigs/kubebuilder/issues/2782

**Comment by jmrodri** [comment]
Based on the error in comment [3 above](https://github.com/operator-framework/operator-sdk/issues/6046#issuecomment-1265853162) that looks like the `manifests` plugin doesn't know about the file `config/manager/controller_manager_config.yaml`. But based on comment [4 above](https://github.com/operator-framework/operator-sdk/issues/6046#issuecomment-1265857089) the component config is supposed to be *optional*, as the Kubebuilder issue states "Meta - ComponentConfig scaffolds should not be done by default".

**Comment by jmrodri** [comment]
@laxmikantbpandhare I think what I would try first is using a branch of operator-sdk that contains the updated kubebuilder in it. Do the following:

**Comment by laxmikantbpandhare** [comment]
Yes, I tried the above command and got the same error on the terminal.

```
FATA[0000] failed to initialize project: unable to scaffold with "base.helm.sdk.operatorframework.io/v1": error updating init manifests: stat config/manager/controller_manager_config.yaml: no such file or directory 
```

Then, I tried below command and it worked.

```
operator-sdk init --plugins helm --domain example.com --group cache --version v1alpha1 --kind Memcached --component-config --helm-chart ../../../hack/generate/samples/internal/helm/testdata/memcached-0.0.2.tgz
```

```
laxmikantbhaskarpandhare@lpandhar-mac check % ls
config
laxmikantbhaskarpandhare@lpandhar-mac check % tree config 
config
├── default
│   ├── kustomization.yaml
│   ├── manager_auth_proxy_patch.yaml
│   └── manager_config_patch.yaml
├── manager
│   ├── controller_manager_config.yaml
│   ├── kustomization.yaml
│   └── manager.yaml
├── prometheus
│   ├── kustomization.yaml
│   └── monitor.yaml
└── rbac
    ├── auth_proxy_client_clusterrole.yaml
    ├── auth_proxy_role.yaml
    ├── auth_proxy_role_binding.yaml
    ├── auth_proxy_service.yaml
    ├── kustomization.yaml
    ├── leader_election_role.yaml
    ├── leader_election_role_binding.yaml
    ├── role_binding.yaml
    └── service_account.yaml
 ```

**Comment by laxmikantbpandhare** [comment]
I found the bug on Kubebuilder which was missed as part of the `--component-config` flag enablement process.

Issue - https://github.com/kubernetes-sigs/kubebuilder/issues/3009
PR - https://github.com/kubernetes-sigs/kubebuilder/pull/3010

**Comment by jmrodri** [comment]
@laxmikantbpandhare okay, that makes sense. I think we need to get that merged into kubebuilder and do a point release of kubebuilder. Then bring that into SDK.

**Comment by laxmikantbpandhare** [comment]
Do you mean patch release? Yes, the minor release is needed for sure.

**Comment by laxmikantbpandhare** [comment]
This is fixed as part of https://github.com/operator-framework/operator-sdk/pull/6047
